### PR TITLE
New Naval Monsters

### PR DESCRIPTION
**Pull request by DoctorBoomstick**
#### Summary
Content "Adds a few new naval themed zombies and feral humans to the aircraft carrier map."

#### Purpose of change
Continuing work started in PR #58049 , where new naval professions were added, this PR aims to make the aircraft carrier start and general location more unique, realistic, and varied. Previously, the carrier’s spawn group had included within it regular feral humans and military/ civilian zombies. On the first count, I felt like these feral humans, the axe, crowbar, pipe, hammer, and spear wielding feral humans, were, from a realism perspective, rather out of place to be abord a carrier which was on active deployment. Firstly, many of these enemies have a rock throwing capability, which doesn’t make sense for the context of a carrier due to the fact that there’s a slight shortage of large chunks of rock present on a military boat. Secondly, when killed all these enemies constantly drop civilian clothing. If one is to assume that the ship was on active deployment, as described within the scenario description, it would be safe to conclude that many of the people abord the ship would be naval personnel, marines or sailors. As such, it doesn’t make sense for such a large number of people abord to be dropping fully civilian gear and clothing all the time. On account of zombies, the second issue carries into the zombie contingent abord the ship as well, and whilst having stock standard zombie soldiers and uniformed zombies is perfectly fine and serviceable, I thought that it’d be fun to add a few specifically naval themed undead.
#### Describe the solution
I’ve added a fair few feral humans and zombies to replace the current aircraft carrier’s contingent of enemies.
The feral humans have been replaced with 7 new types of feral enemies, the first of which being four different flavours of crewmen. In no particular order, they’re armed with wrenches, crash axes, lug wrenches, and mops. When killed, they always drop these items, along with navy related clothing, a small chance for a pistol, and a common workshop tool to reflect their roll abord the carrier. Secondly, I’ve included a feral Marine, which is in essence a replacement for both the cunning and soldier feral humans. This enemy is referenced as having an empty rifle which has a bayonet attached to it. this gives the enemy a two tile ranged stab attack identical to the cunning feral, and they always drop a weapon with a fixed bayonet when killed. Apart for this they also drop standard military equipment. Sixth of all I’ve created a feral officer, which is an enemy that’s armed with a standard M18 containing three rounds and that always drops when the officer is killed. Last of all for feral humans, I’ve created the twisted leatherneck which is an upgraded feral marine. The concept behind this enemy is that the blob is exerting some mutagenic effects on human hosts, similar to what has been seen in fauna, and has accordingly made some bodily changes. Gameplay wise, they’re basically a feral take on the zombie brute.
When it comes to zombies, I’ve added the zombie mariner which is essentially a zombified crewmen and has identical death drops. The devil dog has also been added, which is a zombie marine, along side the zombie officer which, surprisingly enough, is the zombified corpse of a naval officer.
These enemies require the eight point cap that was added in PR #58049 in order to function without throwing up an error, as it is part of their death drops.

#### Describe alternatives you've considered
Simply leaving the enemy groups the way that they currently are, or toning down on the amount of feral humans that are present.
#### Testing
Jumped into an aircraft carrier start with the changes made in #58049 , and wondered around to observe density and overall challenge of the new enemies.

### CODE REVIEW

File: data/json/monsters/feral_humans.json [hunk 1]
```diff
@@ -850,5 +825,224 @@
...
+    "melee_dice_sides": 3,
+    "death_drops": "mon_feral_sailor_lug_wrench_death_drops"
+  },
+  {
+    "id": "mon_feral_marine_bayonet",
+    "type": "MONSTER",
+    "name": { "str": "feral marine" },
+    "description": "This twitching, bloodied person is still clad in a ragged naval uniform, the insignia of the marines visible on their coat and a corresponding tattoo on their forearm.  Clutching a rifle in their quivering hands, you see them repeatedly pull the weapon’s trigger, constantly muttering incoherently and yanking back on the gun’s charging handle every time the rifle fails to fire.  Seemingly having lost the capability to comprehend that their weapon is empty, you’d feel relatively safe were it not for the gleaming bayonet still attached to the end of their firearm.",
```
Comment (TheShadowFerret) [inline data/json/monsters/feral_humans.json:894]: The drops for this include shotguns, which are not what the description says it's holding.

File: data/json/monsters/feral_humans.json [hunk 2]
```diff
@@ -850,5 +825,224 @@
...
+    "death_drops": "mon_feral_sailor_death_drops",
+    "zombify_into": "mon_zombie_sailor",
+    "anger_triggers": [ "FRIEND_DIED", "FRIEND_ATTACKED", "HURT" ],
+    "flags": [ "SEES", "HEARS", "SMELLS", "WARM", "BASHES", "GROUP_BASH", "HUMAN", "CAN_OPEN_DOORS", "PATH_AVOID_DANGER_1" ]
+  },
+  {
+    "id": "mon_feral_sailor_axe",
+    "type": "MONSTER",
+    "copy-from": "mon_feral_sailor_wrench",
+    "description": "Spattered from head to toe in large swaths of dried viscera, the deep blue of this person’s tattered naval uniform has been stained the colour of rust.  Having seemingly had enough presence of mind to consider arming itself, this former crewmen carries a crash axe in its shaking hands.  The vanes on the back of their neck are visibly bulging, their eyes bloodshot, and a mass of crisscrossing blackened vanes stand out against the pale skin of their face as they regard the area around them for perceived threats.",
```
Comment (TheShadowFerret) [inline data/json/monsters/feral_humans.json:870]:
```suggestion
    "description": "Spattered from head to toe in large swaths of dried viscera, the deep blue of this person’s tattered naval uniform has been stained the color of rust.  Having seemingly had enough presence of mind to consider arming itself, this former crewmen carries a crash axe in its shaking hands.  The veins on the back of their neck are visibly bulging, their eyes bloodshot, and a mass of crisscrossing blackened veins stand out against the pale skin of their face as they regard the area around them for perceived threats.",
```

File: data/json/monsters/feral_humans.json [hunk 3]
```diff
@@ -850,5 +825,224 @@
     "extend": { "flags": [ "WATER_CAMOUFLAGE", "NO_BREATHE" ] },
     "delete": { "flags": [ "SMELLS" ] },
     "death_drops": "feral_diver_death_drops_knife"
+  },
+  {
+    "id": "mon_feral_sailor_wrench",
+    "type": "MONSTER",
+    "name": { "str": "feral crewmen" },
+    "description": "clad in a grimy and tattered naval uniform, this former sailor clutches a massive wrench as some form of makeshift weapon.  At a distance, you could mistake them as being another survivor.  However, upon closer inspection you make note of how their veins are visibly bulging, their wide open eyes have a primal, animalistic quality to their gaze, and their breaths come in ragged, wheezing pants.  Whatever the Cataclysm has done to the military crewmen that this twitching shell of their former self once was, you’re certain that their not fully human anymore.",
```
Comment (TheShadowFerret) [inline data/json/monsters/feral_humans.json:833]:
```suggestion
    "description": "Clad in a grimy and tattered naval uniform, this former sailor clutches a massive wrench as some form of makeshift weapon.  At a distance, you could mistake them as being another survivor.  However, upon closer inspection you make note of how their veins are visibly bulging, their wide open eyes have a primal, animalistic quality to their gaze, and their breaths come in ragged, wheezing pants.  Whatever the Cataclysm has done to the military crewmen that this twitching shell of their former self once was, you’re certain that they're not fully human anymore.",
```

**Review by TheShadowFerret — COMMENTED** [review]
Your changes need to be rebased, as they undo #58014 and #58020

**Comment by DoctorBoomstick** [comment]
@ @TheShadowFerret, thanks as always for the suggested changes and fixes. I’ve currently not got access to my main PC, so once I do I’ll implement the other tweaks, ie, removing the shotgun drop entry and fixing up some of the monster armour values which got left out during bug testing.

File: data/json/monsters/feral_humans.json [hunk 4]
```diff
@@ -850,5 +825,224 @@
     "extend": { "flags": [ "WATER_CAMOUFLAGE", "NO_BREATHE" ] },
     "delete": { "flags": [ "SMELLS" ] },
     "death_drops": "feral_diver_death_drops_knife"
+  },
+  {
+    "id": "mon_feral_sailor_wrench",
+    "type": "MONSTER",
+    "name": { "str": "feral crewmen" },
```
Comment (Maleclypse) [inline data/json/monsters/feral_humans.json:832]:
```suggestion
    "name": { "str": "feral crewman", "str_pl": "feral crewmen" },
```

**Comment by fairyarmadillo** [comment]
Don't spontaneously mutating ferals contradict established lore? I thought animals above dog size need actual mutagen to transform, and ferality is meant as just a stop on the way to zombification.

**Comment by DoctorBoomstick** [comment]
@fairyarmadillo My thought process was that, we’ve got stuff like the dear/ spider hybrids and the beaver mutations running around, so perhaps the Blob could also do something in the same style to humans. Though, I get your point, and I must admit I'm not well versed on the differences between ferals/ mutent animals and zombies. You think the Twisted Leatherneck would be better served being switched to just an upgraded form of the zombie marine?

**Comment by fairyarmadillo** [comment]
@DoctorBoomstick The spideer et al are not feral. They are in the mutant faction and are hostile to zombies, which is why you see so many wasp v zombie wars in town. Ferals are in the zombie faction and are not mutants.

The only exception that I can think of are the basement roaches and spiders, which should be in the mutant faction but are flagged as neutral to zombies because otherwise they'd all just kill each other before the player ever showed up. I believe Venera has stated that roaches and spiders are very low priority targets for the blob (which can't zombify them and doesn't really care) and those bugs are clever enough to avoid the zombies.

Adding a mutant feral would create a precedent which would throw the doors open for people to go hog wild with the idea, so I don't think it should be done without somebody at the top acknowledging that and saying yes to it.

**Comment by DoctorBoomstick** [comment]
@fairyarmadillo Alright then, thanks a bunch for the clarifications. When I get a few minutes of time together I’ll modify it to be an upgraded zombie. It should be a relatively simple procedure, just take the leatherneck, slap it into the zombie soldiers file, and arrange the description a little. Thanks again for pointing this inconsistency out.

**Comment by GuardianDll** [comment]
now you need to rebase the PR to hook up the last changes from the game, and to fix all issues test see now

**Comment by anothersimulacrum** [comment]
As a point of note, feral mutation is a thing:
[Discord link](https://discord.com/channels/598523535169945603/598535827169083403/972935259584868382)
> **kevingranade:** [response to ongoing conversation] mutation stuff should not be common at all. The materials are pretty readily available just because of the way our spawning logic works in legacy code and content but you should almost never encounter another mutant especially one that keeps diving deeper and deeper into mutations
> **I-am-erk:** On that note though I would like to still have people who mutate randomly just as animals do
> **I-am-erk:** Yusuke and Makaila in the refugee center for example
> **I-am-erk:** It can still be quite rare, especially for non-ferals. An interesting point might be exploring why ferals often mutate and non ferals rarely do.
> **kevingranade:** Yes sure but more as a risk than an intentional transhumanism thing
> **I-am-erk:** Also lends itself to some in universe anti-mutant sentiment since for every makaila there's ten frenzied cannibal monsters

[Discord link too](https://discord.com/channels/598523535169945603/610001025755119630/857007966464901191)
> **I-am-erk:** Most of the time [the exodii] just shoot mutants on sight, once we have systems for that.
> **I-am-erk:** A highly mutated player will be unable to talk to exodii regardless of faction standing.
> **I-am-erk:** That's not even prejudice or anything, they have good reason for that as we'll see once feral evolutions start being a thing
> **actual-nh:** There's too much blob in you?
> **I-am-erk:** Nah, evolved ferals will be similar to threshold mutants, and dangerous on a scale with or exceeding advanced zombies.

**Comment by fairyarmadillo** [comment]
That sounds like they would be exceptional enemies which would need to be approached with some intentionality.

**Pull request merged**